File: pdf_parser/read_pdf.py
import sys
import os
from pypdf import PdfReader

def read_pdfs():
    dir_path = '../'
    files = [f for f in os.listdir(dir_path) if f.endswith('.pdf')]
    for f in files:
        print(f"\n\n--- CONTENT OF {f} ---")
        try:
            reader = PdfReader(os.path.join(dir_path, f))
            text = ""
            for page in reader.pages:
                text += page.extract_text() + "\n"
            print(text)
        except Exception as e:
            print(f"Error reading {f}: {e}")

if __name__ == '__main__':
    read_pdfs()

File: pdf_parser/read_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

async function readAll() {
  const dir = '../';
  const files = fs.readdirSync(dir).filter(f => f.endsWith('.pdf'));
  for (const f of files) {
    console.log(`\n\n--- CONTENT OF ${f} ---`);
    const dataBuffer = fs.readFileSync(dir + f);
    try {
      const data = await pdf(dataBuffer);
      console.log(data.text);
    } catch(err) {
      console.error(err);
    }
  }
}
readAll();
